src/cai-block.ads

```diff
@@ -1,56 +1,121 @@
+--
+--  @summary Block interface declarations
+--  @author  Johannes Kliemann
+--  @date    2019-04-10
+--
+--  Copyright (C) 2019 Componolit GmbH
+--
+--  This file is part of ada-interface, which is distributed under the terms of the
+--  GNU Affero General Public License version 3.
+--
 
-with System;
 private with Cai.Internal.Block;
 
 pragma Warnings (Off, "type ""Buffer"" is not referenced");
 --  Buffer is only provided to be used in child packages
 
 generic
+   --  Buffer element type, must be 8bit in size
    type Byte is (<>);
+   --  Buffer index type
    type Buffer_Index is range <>;
+   --  Buffer type to be used with all operations of this instance
    type Buffer is array (Buffer_Index range <>) of Byte;
 package Cai.Block
    with SPARK_Mode
 is
-   pragma Compile_Time_Error (Byte'Size /= 8,
-                              "Byte must have a size of 8 bit.");
+   pragma Compile_Time_Error (Byte'Size /= 8, "Byte must have a size of 8 bit.");
 
+   --  Number of Bytes
    type Byte_Length is range 0 .. 2 ** 63 - 1
       with Size => 64;
+   --  Block Id
    type Id is mod 2 ** 64
       with Size => 64;
+   --  Amount of blocks
    type Count is range 0 .. 2 ** 63 - 1
       with Size => 64;
+   --  Size of a block in bytes
    type Size is range 0 .. 2 ** 32 - 1
       with Size => 64;
 
+   --  * Operator for block count and size
```

Comment:
The `*` looks like a bullet point here. Maybe change to 'Operator "*" for block count and size'. Also check similar occurrences below.